runtime/eventlog.c

```diff
@@ -41,32 +41,80 @@
+/* Description of a user event for the 'metadata' stream */
+const char_os *user_event_format = "\nevent {\n     id = %d;\n     name = \"%s\";\n     stream_id = 0;\n     fields := struct {\n        uint8_t span_type;\n     };\n};\n";
+static struct ctf_metadata_header metadata_header = {
+  CTF_METADATA_MAGIC,
+  CTF_TRACE_UUID,
+  0,
+  0,
+  0,
+  0,
+  0,
+  0,
+  CTF_MAJOR_VERSION,
+  CTF_MINOR_VERSION
+};
+
+/* Header at the beginning of every event in the 'data' stream */
 #pragma pack(1)
 struct ctf_event_header {
   uint64_t timestamp;
-  uint32_t pid;
```

Comment:
Why has `pid` been removed?

Reply:
`pid` has been removed from the event header and instead added as packet context. This makes it accessible to the Python extractor that uses the babeltrace2 library. @engil was also of the opinion that having `pid` in the packet context is the 'right' way to do this